File: src/ProjectR/content/scripts/spells/UltimateTaboo.lua
function GetTargetType()
    return Enemies
end

function GetName()
return "Ultimate Taboo"
end

function GetDescription()
return "Drain energy from your allies to empower your strike.\nUses 10%% of max HP per target hit from each ally."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return 2
end

function GetDelay()
  return 0
end

function GetSpellType()
  return Composite
end

function GetMasteries()
  return { FIR }
end

function SpellEffect()
  extraDamage = 0
  ForEachAttackerParty("AddExtraDamage")

  local damage = ((4.5 * aAD + 4.5 * aMD) * (aFIR/100) - (.75 * dDEF + .75 * dMR)) * (100/dFIR) + extraDamage
  d:TakeDamage(damage)
end

function AddExtraDamage(char)
  char:SetTurnCounter(0)
  local extraValue = GetStats(char):GetTotalStat(HP) * .1
  extraDamage = extraDamage + extraValue
end

File: src/ProjectR.Scripting/content/scripts/spells/UltimateTaboo.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Spells
{
    public class UltimateTaboo : SpellScriptBase
    {
        public override string Name { get { return "Ultimate Taboo"; } }
        public override string Description { get { return "Drain energy from your allies to empower your strike.\nExhausts all allies and turns 10%% of their HP into extra damage."; } }

        public override TargetType TargetType { get { return TargetType.Enemies; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { EleMastery.FIR, }; } }
        public override SpellType SpellType { get { return SpellType.Composite; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 200; } }
        public override double Delay { get { return 0; } }

        public override void SpellEffect(ICharacter caster, IList<ICharacter> targets)
        {
            var extraDamage = 0d;
            foreach (var character in GetCasterParty())
            {
                character.TurnCounter = 0;
                var extraValue = HP(character) * .1;
                extraDamage += extraValue;
                DealDamage(character, extraValue);
            }

            foreach (var target in targets)
            {
                Target = target;
                var damage = ((4.5 * aAD + 4.5 * aMD) * (aFIR / 100) - (.75 * dDEF + .75 * dMR)) * (100 / dFIR) + extraDamage;
                DealDamage(damage);
            }
        }
    }
}